### PR TITLE
Identify Huawei Cloud as OpenStack

### PR DESCRIPTION
**Pull request by ixjhuang**
Huawei Cloud is a international cloud provider and it runs OpenStack.
Due to the commit https://github.com/canonical/cloud-init/commit/1efa8a0a030794cec68197100f31a856d0d264ab, we're not able to force the use of the OpenStack datasource.
So we need to become an official provider.

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by archerslaw — APPROVED** [review]
Looks Good To Me.

**Review by archerslaw — APPROVED** [review]
Looks Good To Me.

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thank your for this submission @ixjhuang. I agree to the general approach here, and I also confirm the github username ixjhuang has signed cloud-init's ContributorLicenseAgreement.

In order to accept this PR can you please perform the following:
1. Add trailing comma in the apport.py module
2. Make sure `tox -e do_format` passes on this branch
3. Add a comment to this PR with the output of the file `/run/cloud-init/ds-identify.log` for future reference of the detected platform information  for Huawei

File: tests/unittests/sources/test_openstack.py [hunk 1]
```diff
@@ -603,6 +603,46 @@ def fake_dmi_read(dmi_key):
...
+            if dmi_key == "chassis-asset-tag":
+                return "HUAWEICLOUD"
+            assert False, "Unexpected dmi read of %s" % dmi_key
+
+        m_dmi.side_effect = fake_product_name_dmi_read
+        self.assertTrue(
+            ds.detect_openstack(),
+            "Expected detect_openstack == True on Huawei Cloud VM",
+        )
+
```
Comment (blackboxsw) [inline tests/unittests/sources/test_openstack.py:645]: I'm not sure we this unit test here, as the check for `chassis-asset-tag == HWUAWEICLOUD` will be positive before the check for system-product-name  == "OpenStack Nova" in `detect_openstack`. We already have a `test_detect_openstack_intel_product_name_compute` test that exercises `product == OpenStack Nova` anyway.

File: cloudinit/apport.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "DigitalOcean",
     "E24Cloud",
     "GCE - Google Compute Engine",
+    "Huawei Cloud"
```
[the remark this answers is not in the document]
Comment (ixjhuang) [reply cloudinit/apport.py]: Done.

File: tests/unittests/sources/test_openstack.py (hunk 1 above)
Comment (ixjhuang) [reply tests/unittests/sources/test_openstack.py]: Originally, considering that Huawei cloud has two dmi_key, I wrote test cases respectively.
Agree with your opinion, this test case has been deleted.

File: cloudinit/apport.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ixjhuang) [reply cloudinit/apport.py]: Hello, thanks for your review @blackboxsw .
The code has been modified and I would like to confirm the third comment. In the output of file ```/run/cloud-init/ds-identify.log``` already contains Huawei cloud information, do you mean to add some additional information?
image

[the remark this answers is not in the document]
Comment (blackboxsw) [reply cloudinit/apport.py]: @ixjhuang this is perfect thank you. I wanted to see a point of reference for what ds-identify "sees" on Huawei platform in `DMI_*` values. to confirm other environmental factors available on the platform.

**Review by blackboxsw — COMMENTED** [review]
Thanks for this @ixjhuang.
Looks like `tox -e do_format` will suggest the following diff which wiould fail our CI
After this diff is applied I'll click the CI runner and this PR should be good to merge.
We may hold it a couple of days before merging because we are in the middle of a public release validation of version 22.3 and want to keep commits in tip of main stable for a couple of days to avoid release branch churn. Other than the formatting change. this looks good to me.

```diffdiff --git a/cloudinit/sources/DataSourceOpenStack.py b/cloudinit/sources/DataSourceOpenStack.py
index c252eaae8..88e4f6c2e 100644
--- a/cloudinit/sources/DataSourceOpenStack.py
+++ b/cloudinit/sources/DataSourceOpenStack.py
@@ -32,9 +32,13 @@ DMI_ASSET_TAG_OPENTELEKOM = "OpenTelekomCloud"
 # See github.com/sapcc/helm-charts/blob/master/openstack/nova/values.yaml
 # -> compute.defaults.vmware.smbios_asset_tag for this value
 DMI_ASSET_TAG_SAPCCLOUD = "SAP CCloud VM"
-DMI_ASSET_TAG_HUAWEICLOUD = 'HUAWEICLOUD'
+DMI_ASSET_TAG_HUAWEICLOUD = "HUAWEICLOUD"
 VALID_DMI_ASSET_TAGS = VALID_DMI_PRODUCT_NAMES
-VALID_DMI_ASSET_TAGS += [DMI_ASSET_TAG_HUAWEICLOUD, DMI_ASSET_TAG_OPENTELEKOM, DMI_ASSET_TAG_SAPCCLOUD]
+VALID_DMI_ASSET_TAGS += [
+    DMI_ASSET_TAG_HUAWEICLOUD,
+    DMI_ASSET_TAG_OPENTELEKOM,
+    DMI_ASSET_TAG_SAPCCLOUD,
+]
 
 
 class DataSourceOpenStack(openstack.SourceMixin, sources.DataSource):
```

File: tests/unittests/sources/test_openstack.py (hunk 1 above)
Comment (blackboxsw) [reply tests/unittests/sources/test_openstack.py]: Thanks

**Comment by ixjhuang** [comment]
> 
Thanks for your review and answers @blackboxsw .
The execution results of ```tox -e do_format``` are as follows:
image

**Review by blackboxsw — APPROVED** [review]
LGTM!. thanks for the discussion here

**Pull request merged**